src/services/shiftRotaService.ts

```diff
@@ -77,4 +78,78 @@ export default class ShiftRotaService {
         if (successCount != shiftData.length) console.log(`${shiftData.length - successCount} error/s occured during the process! Check your data sheet.`)
         return;
     }
-}
\ No newline at end of file
+}
+
+//to convert shift data hours to UTC using UTCTimeConversion function
+let passHoursToUTCTimeConversion = (shiftData:ShiftRota[],offset:number) =>{
+    for (let i = 0; i < shiftData.length; i++) {
+        for (const key in shiftData[i]) {
+          if (key === "date") continue;
+    
+          let time = shiftData[i][key];
+    
+          //for empty times like ""
+          if (time === "") continue;
+          //for times like "9-13;15-16"
+          else if (time.split("").includes(";")) {
+            let subTimes = time.split(";");
+            let convertedSubTimes: string[] = [];
+            subTimes.forEach((time) => {
+              let [startTimeLocal, endTimeLocal]: string[] = time.split("-");
+              let convertedTime = UTCTimeConversion(startTimeLocal, endTimeLocal,offset);
+              convertedSubTimes.push(convertedTime);
+            });
+    
+            shiftData[i][key] = convertedSubTimes.join(";");
+          }
+          //for times like "9-17"
+          else {
+            let [startTimeLocal, endTimeLocal]: string[] = time.split("-");
+            let convertedTime = UTCTimeConversion(startTimeLocal, endTimeLocal,offset);
+    
+            shiftData[i][key] = convertedTime;
+          }
+        }
+      }
+      return shiftData
+}
+
+
+//UTC time conversion
+const UTCTimeConversion = (startTimeLocal: string, endTimeLocal: string,offset:number) => {
+    function convertMinsToHrsMins(a: number) {
+      let hours = Math.trunc(a / 60);
+      let minutes = a % 60;
+      //If offset is full hours then result 8-16, if not 8:15-16:15
+      if (minutes > 0) return hours + ":" + minutes;
+      else return hours;
+    }
+  
+    // //Calculating offset(time diff bw local and UTC time)
+    // let offset: number = new Date().getTimezoneOffset();
+  
+    //Converting start time to UTC
+    let startTimeLocalMinutes: number;
+    if (startTimeLocal.length > 2)
+      startTimeLocalMinutes =
+        Number(startTimeLocal.split(":")[0]) * 60 +
+        Number(startTimeLocal.split(":")[1]);
+    else startTimeLocalMinutes = +startTimeLocal * 60;
+  
+    let startTimeLocalUTCMinutes: number = startTimeLocalMinutes + offset;
+    let startTimeUTC = convertMinsToHrsMins(startTimeLocalUTCMinutes);
+  
+    //Converting end time to UTC
+    let endTimeLocalMinutes: number;
+    if (endTimeLocal.length > 2)
+      endTimeLocalMinutes =
+        Number(endTimeLocal.split(":")[0]) * 60 +
+        Number(endTimeLocal.split(":")[1]);
+    else endTimeLocalMinutes = +endTimeLocal * 60;
+  
+    let endTimeLocalUTCMinutes: number = endTimeLocalMinutes + offset;
+    let endTimeUTC = convertMinsToHrsMins(endTimeLocalUTCMinutes);
+  
+    return `${startTimeUTC}-${endTimeUTC}`;
+  };
```

Comment:
I wonder if it wouldn't be simplier to convert each 'timestamp' separately in the passHoursToUTCTimeConversion  function - so in passHoursToUTCTimeConversion you additionally split hours by '-' and pass them to UTCTimeConversion, and in UTCTimeConversion you just convert one 'hour' not start/end date structure

It's not a requirement though, but probably such code would be simplier/shorter.

This is just a note, give it a try if you want, if you feel the current solution is ok you can close it

Reply:
As discussed, not proceeding with this approach.

However, this function had a lot of repeated lines - I have created one function (convertLocalTimeToUTC) and calling it to not repeat anymore - following this DRY approach has decreased a considerable number of lines.